From c37419e459bd5863534719fa50755174d0912e6d Mon Sep 17 00:00:00 2001
From: Romain Naour <romain.naour@gmail.com>
Date: Fri, 30 Apr 2021 14:32:21 +0200
Subject: [PATCH] jitter: jitter-config.in: avoit host poisoning while
 cross-compiling

When jitter is bundled into another project (poke), jitter-config --cppflags
return an unsafe path "-I /usr/include".

In the context of Buildroot, such configuration script woulf be "fixed"
after installation. But jitter-config is not installed by poke since it's
build as static library and liked to poke.

For now, patch jitter-config.in instead of jitter-config.in.m4sh since
there is an issue while converting the M4sh m4sh script ( .in.m4sh )
into a portable shell script ( .in ) ready to be processed by aclocal
for @-substitutions.

[1] https://git.buildroot.net/buildroot/tree/package/pkg-generic.mk?h=2021.02.1#n291

Signed-off-by: Romain Naour <romain.naour@gmail.com>
---
 jitter/bin/jitter-config.in | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/jitter/bin/jitter-config.in b/jitter/bin/jitter-config.in
index 5e124f0..af0ac9d 100644
--- a/jitter/bin/jitter-config.in
+++ b/jitter/bin/jitter-config.in
@@ -998,11 +998,7 @@ while test "$#" != "0"; do
             append_to_output cflags;;
         --cppflags)
             no_option_argument
-            append_to_output cppflags
-            # Append a -I argument.  This is defined separately from the rest,
-            # as the installation prefix can be decided very late, at Jitter
-            # installation time.
-            output="$output -I $includedir";;
+            append_to_output cppflags;;
         --ldadd)
             no_option_argument
             append_to_output ldadd;;
-- 
2.30.2

